### PR TITLE
Migrate System Tests to Chai

### PR DESCRIPTION
**Pull request by bharathr145**
- Use chai expect assertions
- Added global expect for System test

### CODE REVIEW

File: test/system/appveyor-yml.test.js [hunk 1]
```diff
@@ -46,20 +43,23 @@ describe('.appveyor.yml', function () {
         });
 
         it('should have correct install scripts', function () {
-            expect(appveyorYAML.install[0].ps).to.be('Install-Product node $env:nodejs_version');
-            expect(appveyorYAML.install[1]).to.be('npm cache clean --force');
-            expect(appveyorYAML.install[2]).to.be('appveyor-retry npm install');
+            expect(appveyorYAML.install[0].ps).to.equal('Install-Product node $env:nodejs_version');
```
Comment (kunagpal) [inline test/system/appveyor-yml.test.js]: https://www.chaijs.com/api/bdd/#nested

File: test/system/appveyor-yml.test.js [hunk 2]
```diff
@@ -20,12 +17,12 @@ describe('.appveyor.yml', function () {
...
 
     it('should be a valid yml', function () {
-        expect(appveyorYAMLError && appveyorYAMLError.message || appveyorYAMLError).to.not.be.ok();
+        expect(appveyorYAMLError && appveyorYAMLError.message || appveyorYAMLError).to.be.undefined;
     });
 
     describe('structure', function () {
         it('should have an init script', function () {
-            expect(appveyorYAML.init[0]).to.be('git config --global core.autocrlf input');
+            expect(appveyorYAML.init[0]).to.equal('git config --global core.autocrlf input');
```
Comment (kunagpal) [inline test/system/appveyor-yml.test.js]: https://www.chaijs.com/api/bdd/#nested

File: test/system/appveyor-yml.test.js [hunk 3]
```diff
@@ -46,20 +43,23 @@ describe('.appveyor.yml', function () {
...
+                deploy: 'off'
+            });
         });
 
-        it('should have notifications must be configured correctly', function () {
-            expect(appveyorYAML.notifications).to.be.an(Array);
-            expect(appveyorYAML.notifications[0].provider).to.be('Slack');
-            expect(appveyorYAML.notifications[0].incoming_webhook.secure).to.be.ok();
+        it('should have notifications configured correctly', function () {
+            expect(appveyorYAML.notifications).to.be.an('array');
```
Comment (kunagpal) [inline test/system/appveyor-yml.test.js]: Switch to a deep `eql` assertion instead.

File: test/system/nsp.test.js [hunk 4]
```diff
@@ -2,10 +2,8 @@
  * @fileOverview Ensures nsprc is as expected
  */
 
-var fs = require('fs'),
-    expect = require('expect.js');
+var fs = require('fs');
```
Comment (kunagpal) [inline test/system/nsp.test.js]: **A side note:** we'll be removing all NSP related files down the line.

File: test/system/appveyor-yml.test.js [hunk 5]
```diff
@@ -1,6 +1,3 @@
-/* global describe, it */
-var expect = require('expect.js');
-
 describe('.appveyor.yml', function () {
     var fs = require('fs'),
```
Comment (codenirvana) [inline test/system/appveyor-yml.test.js]: Move `require`s at the top.

File: test/system/jsdoc-config.test.js [hunk 6]
```diff
@@ -3,11 +3,9 @@
  * on the content of the file as well. Any change to .jsdoc-config.json must be accompanied by valid test case in this
  * spec-sheet.
  */
-/* global describe, it */
+
 describe('JSDoc configuration', function () {
     var fs = require('fs'),
```
Comment (codenirvana) [inline test/system/jsdoc-config.test.js]: Move `require` at the top.

File: test/system/repository.test.js [hunk 7]
```diff
@@ -3,10 +3,8 @@
  * content of the file as well. Any change to package.json must be accompanied by valid test case in this spec-sheet.
  */
 var _ = require('lodash'),
-    expect = require('expect.js'),
     parseIgnore = require('parse-gitignore');
 
-/* global describe, it */
 describe('project repository', function () {
     var fs = require('fs');
```
Comment (codenirvana) [inline test/system/repository.test.js]: Move `require` at the top.

File: test/system/travis-yml.test.js [hunk 8]
```diff
@@ -1,6 +1,3 @@
-/* global describe, it */
-var expect = require('expect.js');
-
 describe('travis.yml', function () {
     var fs = require('fs'),
```
Comment (codenirvana) [inline test/system/travis-yml.test.js]: Move `require`s at the top.

**Pull request merged**